File: app/src/main/java/recycler/ActRecycler.java
package recycler;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.example.exemplo.R;

public class ActRecycler extends AppCompatActivity {

    RecyclerView recycler;
    RecyclerView.LayoutManager layoutManager;
    AdapterChamada adapter;
    String[] chamada = {
            "Rafinha",
            "Henrique",
            "Xandão",
            "Daniel",
            "Peter Henry",
            "404",
            "Giulia",
            "Lucas",
            "Eduardo",
            "Sandro",
            "Felipe",
            "Gabriel"};

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.act_recycler);
        recycler = findViewById(R.id.recycler);
        layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
        adapter = new AdapterChamada(chamada);

        recycler.setLayoutManager(layoutManager);
        recycler.setAdapter(adapter);
    }
}

File: app/src/main/java/recycler/ActRecycler.kt
package recycler

import android.os.Bundle
import android.widget.Toast
import android.widget.Toast.LENGTH_LONG
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager.VERTICAL
import androidx.recyclerview.widget.RecyclerView

class ActRecycler : AppCompatActivity(), OnClickAluno {

    private val recycler by lazy {
        RecyclerView(this)
    }
    private val manager by lazy {
        LinearLayoutManager(this, VERTICAL, false)
    }

    private val alunos = listOf(
        "Rafinha",
        "Henrique",
        "Xandão",
        "Daniel",
        "Peter Henry",
        "Giulia",
        "Felipe"
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(recycler)

        recycler.layoutManager = manager
        recycler.adapter = AdapterAlunos(this, alunos)
    }

    override fun onClickAluno(position: Int) {
        Toast.makeText(this, alunos[position], LENGTH_LONG).show()
    }
}


